src/content_scripts/autocorrect.js

```diff
@@ -0,0 +1,400 @@
+            } else if (insert === '"') {
+                insert = re.test(prevouschar) ? "“" : "”";
+            }
+            deletecount = 1;
+            output = true;
+        }
+        const prevoustext = value.slice(caretposition < (longest + 1) ? 0 : caretposition - (longest + 1), caretposition - 1);
+        const regexResult = symbolpatterns.exec(prevoustext);
+        // Autocorrect Unicode Symbols
+        if (regexResult) {
+            const text = value.slice(caretposition < longest ? 0 : caretposition - longest, caretposition);
+            const aregexResult = symbolpatterns.exec(text);
+            const aaregexResult = apatterns.exec(text);
+            if (!aaregexResult && (!aregexResult || (caretposition <= longest ? regexResult.index < aregexResult.index : regexResult.index <= aregexResult.index))) {
+                insert = autocorrections[regexResult[0]] + (event.keyCode === 13 ? "\n" : insert);
+                deletecount = regexResult[0].length + 1;
+                output = true;
+            }
+        } else {
+            // Convert fractions and mathematical constants to Unicode characters
+            if (!output && fracts) {
+                // Numbers: https://regex101.com/r/7jUaSP/2
+                const re = /[0-9]+(\.[0-9]+)?$/;
+                const prevoustext = value.slice(0, caretposition - 1);
+                const regexResult = re.exec(prevoustext);
+                if (regexResult) {
+                    const text = value.slice(0, caretposition);
+                    const aregexResult = re.exec(text);
+                    if (!aregexResult) {
+                        const label = outputLabel(regexResult[0], regexResult[1]);
+                        const index = firstDifferenceIndex(label, regexResult[0]);
+                        if (index >= 0) {
+                            insert = label.slice(index) + (event.keyCode === 13 ? "\n" : insert);
+                            deletecount = regexResult[0].length - index + 1;
+                            output = true;
+                        }
+                    }
+                }
+            }
+        }
+        if (output) {
+            const text = value.slice(caretposition - deletecount, caretposition);
+            deleteCaret(target, text);
+            insertCaret(target, insert);
+            console.debug("Autocorrect: “%s” was replaced with “%s”.", text, insert);
+
+            insertedText = insert;
+            deletedText = text;
+
+            lastTarget = target;
+            lastCaretPosition = caretposition - deletecount + insert.length;
+        }
+    }
+}
+
+/**
+ * Undo autocorrect.
+ *
+ * @param {Object} event
+ * @returns {void}
+ */
+function undoAutocorrect(event) {
+    // console.log('keyup', event.key, event.key.length, event.keyCode);
+    if (!(!event.ctrlKey && !event.metaKey && !event.altKey)) {
+        return;
+    }
+    // Backspace
+    if (event.keyCode === 8) {
```

Comment:
Hmm, early return here pls…

Invert the logic. If the key is not backspace then return.
Also, if you then do this, you may even merge that "if" with the modifier keys above.